ksql-engine/src/main/java/io/confluent/ksql/engine/InsertValuesExecutor.java

```diff
@@ -149,42 +186,31 @@ public void execute(
       throw new KsqlException("Cannot insert values into windowed stream/table!");
     }
 
-    final KsqlConfig config = statement.getConfig()
-        .cloneWithPropertyOverwrite(statement.getOverrides());
-
     try {
       final RowData row = extractRow(insertValues, dataSource);
       final byte[] key = serializeKey(row.key, dataSource, config, serviceContext);
       final byte[] value = serializeValue(row.value, dataSource, config, serviceContext);
 
       final String topicName = dataSource.getKafkaTopicName();
 
-      final ProducerRecord<byte[], byte[]> record = new ProducerRecord<>(
+      return new ProducerRecord<>(
           topicName,
           null,
           row.ts,
           key,
           value
       );
-
-      producer.sendRecord(record, serviceContext, config.getProducerClientConfigProps());
-    } catch (final TopicAuthorizationException e) {
-      // TopicAuthorizationException does not give much detailed information about why it failed,
-      // except which topics are denied. Here we just add the ACL to make the error message
-      // consistent with other authorization error messages.
-      final Exception rootCause = new KsqlTopicAuthorizationException(
-          AclOperation.WRITE,
-          e.unauthorizedTopics()
-      );
-
-      throw new KsqlException("Failed to insert values into stream/table: "
-          + insertValues.getTarget().name(), rootCause);
-    } catch (final Exception e) {
-      throw new KsqlException("Failed to insert values into stream/table: "
-          + insertValues.getTarget().name(), e);
+    } catch (Exception e) {
+      throw new KsqlStatementException(
+          createInsertFailedExceptionMessage(insertValues) + " " + e.getMessage(),
+          statement.getStatementText(), e);
```

Comment:
nit: code style requires these on separate lines... 

```suggestion
          statement.getStatementText(), 
          e
       );
```